### PR TITLE
[collector] Update internal metrics config examples, suggest OTLP for pushing metrics

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]

File: content/en/docs/collector/internal-telemetry.md [hunk 1]
```diff
@@ -59,30 +64,34 @@ service:
...
 ```
 
-The Collector can also be configured to scrape its own metrics using a
-[Prometheus receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver)
-and send them through configured pipelines. For example:
+The Collector can also be configured to push its own metrics to an
+[OTLP receiver](https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver)
+and send them through configured pipelines. In the following example,
+the Collector is configured to push metrics every 10s using OTLP gRPC to
+`localhost:8888`:
```
Comment (codeboten) [inline content/en/docs/collector/internal-telemetry.md:71]:
```suggestion
`localhost:14317`:
```

File: content/en/docs/collector/internal-telemetry.md [hunk 2]
```diff
@@ -59,30 +64,34 @@ service:
...
-          static_configs:
-            - targets: ['0.0.0.0:8888']
-          metric_relabel_configs:
-            - source_labels: [__name__]
-              regex: '.*grpc_io.*'
-              action: drop
+  otlp/internal_metrics:
+    protocols:
+      grpc:
+        endpoint: localhost:8888
```
Comment (codeboten) [inline content/en/docs/collector/internal-telemetry.md:78]:
```suggestion
        endpoint: localhost:14317
```

File: content/en/docs/collector/internal-telemetry.md [hunk 3]
```diff
@@ -59,30 +64,34 @@ service:
...
       exporters: [debug]
+  telemetry:
+    metrics:
+      readers:
+        - periodic:
+            interval: 10000
+            exporter:
+              otlp/metrics-agent:
+                protocol: grpc/protobuf
+                endpoint: localhost:8888
```
Comment (codeboten) [inline content/en/docs/collector/internal-telemetry.md:94]:
```suggestion
                endpoint: localhost:14317
```

File: content/en/docs/collector/internal-telemetry.md [hunk 4]
```diff
@@ -23,18 +23,23 @@ By default, the Collector exposes its own telemetry in two ways:
 
 You can configure how internal metrics are generated and exposed by the
 Collector. By default, the Collector generates basic metrics about itself and
-exposes them for scraping at `http://127.0.0.1:8888/metrics`. You can expose the
-endpoint to one specific or all network interfaces when needed. For
+exposes them using the otel-go Prometheus exporter for scraping the at
```
Comment (codeboten) [inline content/en/docs/collector/internal-telemetry.md:26]:
```suggestion
exposes them using the OpenTelemetry Go [Prometheus exporter](https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/prometheus) for scraping the at
```

**Review by codeboten — APPROVED** [review]
Recommend changing the OTLP port to 14317 (so as not to confuse the prometheus port) in the example

File: content/en/docs/collector/internal-telemetry.md [hunk 5]
```diff
@@ -23,18 +23,24 @@ By default, the Collector exposes its own telemetry in two ways:
 
 You can configure how internal metrics are generated and exposed by the
 Collector. By default, the Collector generates basic metrics about itself and
-exposes them for scraping at `http://127.0.0.1:8888/metrics`. You can expose the
-endpoint to one specific or all network interfaces when needed. For
-containerized environments, you might want to expose this port on a public
-interface.
+exposes them using the OpenTelemetry Go
+[Prometheus exporter](https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/prometheus)
+for scraping the at `http://127.0.0.1:8888/metrics`. You can expose the endpoint
```
Comment (sfc-gh-mcreech) [inline content/en/docs/collector/internal-telemetry.md:28]:
```suggestion
for scraping at `http://127.0.0.1:8888/metrics`. You can expose the endpoint
```

**Review by tedsuo — APPROVED** [review]

**Review by tiffany76 — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Pull request merged**